src/subclassers.ts

```diff
@@ -34,6 +42,46 @@ type FunctionTaking<TIn> = (param: TIn) => any;
 
 type HasTypedFunctionOn<T, K extends string> = Record<K, FunctionTaking<T>>;
 
```

Comment:
i wasn't sure about i need to add some types here because i think i can do that in HTPipes for this functions